### PR TITLE
 PE-31 | Created Questions table with CRUD Operations

### PR DESCRIPTION
**Pull request by dominikyz**
WHAT I DID:

- Created Questions table
- Added CRUD Operations

TO TEST CRUD OPERATIONS:

1. Clone the repository `project-eureka-spring` 
2. Download postman https://www.postman.com/downloads/.
2. In your terminal, checkout to the branch PE-31-ImplementQuestionsTable :  `git checkout PE-31-ImplementQuestionsTable` 
3. Start spring-boot application : `mvn spring-boot:run`

You should see similar output as here :
image

4. Open Postman test POST : 

- for adding records use POST:
- Screenshot 2021-02-09 230925

URL : `localhost:8080/v1/questions`

paste test record into the body ( you can edit test record to your preference )

`
{
"title" : "TestRecord",
"questionDate" : "2012-04-23T18:25:43.511Z",
"description" : "Description goes here",
"category" : "science",
"status" : 1,
"visible" : 1,
"userId" : 154
}
`

Hit SEND. If everything worked correctly, you'll see reply from database on the bottom
image

5. In Postman test GET:

URL : `localhost:8080/v1/questions`

get

To confirm it works properly, you should see all entries from questions table

6. In Postman test GET for a specific question:

Use same URL and add question id in the end : `localhost:8080/v1/questions/"paste question id here without quotes"`

When you hit send, you should see question info related only to the specified question id

Screenshot 2021-02-10 114156

6. In Postman test PUT for a specific question:

Works only to update existing question. You need question id and update any field.
Here is an example of BEFORE:
before

And here is AFTER with an updated field "description" : 
after

7. In Postman test DELETE for a specific question:

Use same URL and add question id in the end : `localhost:8080/v1/questions/"paste question id here without quotes"`

delete

----UPDATE----
Repeat steps 4-7 to test answers table in Postman  using this : `localhost:8080/v1/answers`
Test payload could be this : 
`
{
    "mediaUrls": [
        "www.google.com"
    ],
    "answerDate":  "2012-04-23T18:25:43.511Z",
    "description": "some description",
    "questionId": 7,
    "userId": 74
}
`
Step 4 should look like this:

post

Step 5 should look like this:

get

Step 6 should look like this:

put

Step 7  should look like this:

delete

### CODE REVIEW

File: src/main/java/com/ProjectEureka/backend/models/Question.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    @Field("media_urls")
+    private List<String> mediaUrls;
+
+    @Field("status")
+    private String status;
+
+    private Boolean visible;
+
+    @Field("user_id")
+    private Long userId;
```
Comment (MaxLian11) [inline src/main/java/com/ProjectEureka/backend/models/Question.java]: I believe user_id must a string, according to the user table. Although Answer table has it incorrect too, just change it in your model only

**Review by MaxLian11 — CHANGES_REQUESTED** [review]

File: src/main/java/com/ProjectEureka/backend/models/Question.java (hunk 1 above)
Comment (MaxLian11) [reply src/main/java/com/ProjectEureka/backend/models/Question.java]: But all in all it works fine, good job!

Comment (dominikyz) [reply src/main/java/com/ProjectEureka/backend/models/Question.java]: Done, good sir!

**Review by MaxLian11 — APPROVED** [review]
Everything works as intended, great job

File: src/main/java/com/ProjectEureka/backend/models/Question.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
+package com.ProjectEureka.backend.models;
+
+import org.springframework.data.annotation.Id;
+import org.springframework.data.mongodb.core.mapping.Document;
+import org.springframework.data.mongodb.core.mapping.Field;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
```
Comment (davidsquines) [inline src/main/java/com/ProjectEureka/backend/models/Question.java:9]: Remove Lombok.getter and lombok.setter as they are already covered in the lombok.data

**Review by davidsquines — APPROVED** [review]
Works as intended. Just fixed the change I put in the comment

File: src/main/java/com/ProjectEureka/backend/models/Question.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+@NoArgsConstructor
+@Document(collection = "Questions")
+public class Question {
+    @Id
+    private String id;
+
+    private String title;
+
+    @Field("question_date")
+    private String questionDate;
```
Comment (tonynguyen98) [inline src/main/java/com/ProjectEureka/backend/models/Question.java]: Ideally, we want this to be of type `Date`. Please also apply it to the answers model.

**Review by tonynguyen98 — APPROVED** [review]
Great work, just one minor change.

File: src/main/java/com/ProjectEureka/backend/models/Question.java (hunk 2 above)
Comment (dominikyz) [reply src/main/java/com/ProjectEureka/backend/models/Question.java:9]: done in the latest commit

File: src/main/java/com/ProjectEureka/backend/models/Question.java (hunk 3 above)
Comment (dominikyz) [reply src/main/java/com/ProjectEureka/backend/models/Question.java]: done in the latest commit

**Review by ayeejane — APPROVED** [review]
Everything works as intended!

**Pull request merged**